Issue

ryan-endacott opened the pull request:
This fixes #216 by making `gb test -v` return correct verbose output without extra debug info.  

Here is some output for these sample tests:
```
package hi

import (
	"testing"
)

func TestSuccess(t *testing.T) {
	t.Log("Success!")
}

func TestFailure(t *testing.T) {
	t.Error("Fail!")
}
```

Before this change:
```
$ gb test -v
DEBUG project root "/home/ryan/.gvm/pkgsets/go1.4rc2/global/src/github.com/constabulary/test"
DEBUG matchPackages: /home/ryan/.gvm/pkgsets/go1.4rc2/global/src/github.com/constabulary/test/src ...
DEBUG args: [hi]
DEBUG loadPackage: hi true (/home/ryan/.gvm/pkgsets/go1.4rc2/global/src/github.com/constabulary/test/src/hi)
DEBUG loadPackage: testing true (/home/ryan/.gvm/gos/go1.4rc2/src/testing)
DEBUG {hi hi /home/ryan/.gvm/pkgsets/go1.4rc2/global/src/github.com/constabulary/test/src/hi} is test scoped, not installing
DEBUG loadTestFuncs: [hi_test.go], []
DEBUG {hi hi/testmain /tmp/gb153125905/hi/_test} is test scoped, not installing
DEBUG scheduling run of [/tmp/gb153125905/hi/testmain/_test/hi.test -test.v]
DEBUG gc:gc hi /home/ryan/.gvm/pkgsets/go1.4rc2/global/src/github.com/constabulary/test /tmp/gb153125905/hi/_test/hi.a [src/hi/hi.go src/hi/hi_test.go]
DEBUG cd /home/ryan/.gvm/pkgsets/go1.4rc2/global/src/github.com/constabulary/test; [/home/ryan/.gvm/gos/go1.4rc2/pkg/tool/linux_amd64/6g -p hi -pack -o /tmp/gb153125905/hi/_test/hi.a -I /tmp/gb153125905 -I /home/ryan/.gvm/pkgsets/go1.4rc2/global/src/github.com/constabulary/test/pkg/linux/amd64 -complete src/hi/hi.go src/hi/hi_test.go]
DEBUG gc:gc hi/testmain /home/ryan/.gvm/pkgsets/go1.4rc2/global/src/github.com/constabulary/test /tmp/gb153125905/hi/testmain/_test/hi/testmain.a [../../../../../../../../../../tmp/gb153125905/hi/_test/_testmain.go]
DEBUG cd /home/ryan/.gvm/pkgsets/go1.4rc2/global/src/github.com/constabulary/test; [/home/ryan/.gvm/gos/go1.4rc2/pkg/tool/linux_amd64/6g -p hi/testmain -pack -o /tmp/gb153125905/hi/testmain/_test/hi/testmain.a -I /tmp/gb153125905/hi/_test -I /tmp/gb153125905 -I /home/ryan/.gvm/pkgsets/go1.4rc2/global/src/github.com/constabulary/test/pkg/linux/amd64 -complete ../../../../../../../../../../tmp/gb153125905/hi/_test/_testmain.go]
DEBUG cd .; [/home/ryan/.gvm/gos/go1.4rc2/pkg/tool/linux_amd64/6l -o /tmp/gb153125905/hi/testmain/_test/hi.test -L /tmp/gb153125905/hi/_test -L /tmp/gb153125905 -L /home/ryan/.gvm/pkgsets/go1.4rc2/global/src/github.com/constabulary/test/pkg/linux/amd64 -extld=gcc /tmp/gb153125905/hi/testmain/_test/hi/testmain.a]
DEBUG run [/tmp/gb153125905/hi/testmain/_test/hi.test -test.v]
=== RUN TestSuccess
--- PASS: TestSuccess (0.00s)
	hi_test.go:8: Success!
=== RUN TestFailure
--- FAIL: TestFailure (0.00s)
	hi_test.go:12: Fail!
FAIL
DEBUG test duration: 211.827632ms map[compile:22.432854ms link:184.906166ms]
FATAL command "test" failed: run [/tmp/gb153125905/hi/testmain/_test/hi.test -test.v]: exit status 1
```

After this change:
```
$ gb test -v
=== RUN TestSuccess
--- PASS: TestSuccess (0.00s)
	hi_test.go:8: Success!
=== RUN TestFailure
--- FAIL: TestFailure (0.00s)
	hi_test.go:12: Fail!
FAIL
FATAL command "test" failed: run [/tmp/gb585619063/hi/testmain/_test/hi.test -test.v]: exit status 1
```

ryan-endacott commented:
I updated the description and am now looking into the failing flag tests.  Do the `TestDeduceRemoteRepo` tests have anything to do with my change?

davecheney commented:
Because those tests pull stuff from the internet they can occasionally fail
if the internet is having a lie down. I'll check it out this evening.

On Thu, 9 Jul 2015 01:19 Ryan Endacott <notifications@github.com> wrote:

> I updated the description and am now looking into the failing flag tests.
> Do the TestDeduceRemoteRepo tests have anything to do with my change?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/constabulary/gb/pull/263#issuecomment-119854085>.
>

ryan-endacott commented:
Yeah I think that's the problem, they were broken on master for me locally
too. I'll work on fixing the testflags tests tonight.

On Thu, Jul 9, 2015, 5:20 AM Dave Cheney <notifications@github.com> wrote:

> Because those tests pull stuff from the internet they can occasionally fail
> if the internet is having a lie down. I'll check it out this evening.
>
> On Thu, 9 Jul 2015 01:19 Ryan Endacott <notifications@github.com> wrote:
>
> > I updated the description and am now looking into the failing flag tests.
> > Do the TestDeduceRemoteRepo tests have anything to do with my change?
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/constabulary/gb/pull/263#issuecomment-119854085>.
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/constabulary/gb/pull/263#issuecomment-119938858>.
>

davecheney commented:
Yes it looks like they broke recently. I'll try to fix them tomorrow during
the hack day.

On Thu, 9 Jul 2015 11:10 Ryan Endacott <notifications@github.com> wrote:

> Yeah I think that's the problem, they were broken on master for me locally
> too. I'll work on fixing the testflags tests tonight.
>
> On Thu, Jul 9, 2015, 5:20 AM Dave Cheney <notifications@github.com> wrote:
>
> > Because those tests pull stuff from the internet they can occasionally
> fail
> > if the internet is having a lie down. I'll check it out this evening.
> >
> > On Thu, 9 Jul 2015 01:19 Ryan Endacott <notifications@github.com> wrote:
> >
> > > I updated the description and am now looking into the failing flag
> tests.
> > > Do the TestDeduceRemoteRepo tests have anything to do with my change?
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <https://github.com/constabulary/gb/pull/263#issuecomment-119854085>.
> > >
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/constabulary/gb/pull/263#issuecomment-119938858>.
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/constabulary/gb/pull/263#issuecomment-120074489>.
>

ryan-endacott commented:
I fixed the flag tests, but I'm not 100% sure my change won't have unintentional side effects.  I changed `q` and `v` to be handled by the test binary.  You may want to make that doesn't break anything.  In particular, the normal debug output isn't given with `-v` anymore.  

Also, no rush on merging this - feel free to do more interesting things during the hack day. Have fun!

davecheney commented:
@ryan-endacott I found the problem, the cert on labix.org expired a few days ago, i've pushed a fix to pull that.

ryan-endacott commented:
Thanks!  I repushed and the tests are now green.

ryan-endacott commented:
Thanks!

On Sun, Jul 12, 2015, 10:11 PM Dave Cheney <notifications@github.com> wrote:

> Merged #263 <https://github.com/constabulary/gb/pull/263>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/constabulary/gb/pull/263#event-354004740>.
>

davecheney commented:
Not at all. Thank you.

On Sun, Jul 12, 2015 at 11:16 PM, Ryan Endacott <notifications@github.com>
wrote:

> Thanks!
>
> On Sun, Jul 12, 2015, 10:11 PM Dave Cheney <notifications@github.com>
> wrote:
>
> > Merged #263 <https://github.com/constabulary/gb/pull/263>.
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/constabulary/gb/pull/263#event-354004740>.
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/constabulary/gb/pull/263#issuecomment-120814978>.
>
